src/librustc/ty/codec.rs

```diff
@@ -393,6 +403,15 @@ macro_rules! implement_ty_decoder {
                     decode_const(self)
                 }
             }
+
+            impl<$($typaram),*> SpecializedDecoder<&'tcx $crate::mir::interpret::Allocation>
```

Comment:
Doesn't this conflict with the `derive` on `Allocation`? IIRC this produces runtime issues if done incorrectly. Have a look at what `AllocId` does

Reply:
I don't think so. This decodes a pointer to an `Allocation`, while the `derive` is for `Allocation` itself.